diff --git a/Tools/wasm/config.site-wasm32-wasi b/Tools/wasm/config.site-wasm32-wasi
index 4a1a466a4a..7ffd6eebd5 100644
--- a/Tools/wasm/config.site-wasm32-wasi
+++ b/Tools/wasm/config.site-wasm32-wasi
@@ -49,3 +49,6 @@ ac_cv_func_preadv=no
 ac_cv_func_readv=no
 ac_cv_func_pwritev=no
 ac_cv_func_writev=no
+
+# WASI SDK 21.0 has libdl
+ac_cv_lib_dl_dlopen=no
diff --git a/Tools/wasm/wasm_build.py b/Tools/wasm/wasm_build.py
index f6e2629656..6fe13ff8f0 100755
--- a/Tools/wasm/wasm_build.py
+++ b/Tools/wasm/wasm_build.py
@@ -485,6 +485,10 @@ def configure_cmd(self) -> List[str]:
             opt = "enable" if self.pthreads else "disable"
             cmd.append(f"--{opt}-wasm-pthreads")
 
+        cmd.append(f"--with-pymalloc")
+        cmd.append(f"--with-computed-gotos")
+        cmd.append(f"--with-lto=thin")
+
         if self.host != Host.build:
             cmd.append(f"--with-build-python={BUILD.python_cmd}")
 
