diff --git a/Include/cpython/pthread_stubs.h b/Include/cpython/pthread_stubs.h
index e542eaa..7236591 100644
--- a/Include/cpython/pthread_stubs.h
+++ b/Include/cpython/pthread_stubs.h
@@ -56,39 +56,39 @@ typedef struct { unsigned __attr; } pthread_attr_t;
 #endif
 
 // mutex
-PyAPI_FUNC(int) pthread_mutex_init(pthread_mutex_t *restrict mutex,
+int pthread_mutex_init(pthread_mutex_t *restrict mutex,
                                    const pthread_mutexattr_t *restrict attr);
-PyAPI_FUNC(int) pthread_mutex_destroy(pthread_mutex_t *mutex);
-PyAPI_FUNC(int) pthread_mutex_trylock(pthread_mutex_t *mutex);
-PyAPI_FUNC(int) pthread_mutex_lock(pthread_mutex_t *mutex);
-PyAPI_FUNC(int) pthread_mutex_unlock(pthread_mutex_t *mutex);
+int pthread_mutex_destroy(pthread_mutex_t *mutex);
+int pthread_mutex_trylock(pthread_mutex_t *mutex);
+int pthread_mutex_lock(pthread_mutex_t *mutex);
+int pthread_mutex_unlock(pthread_mutex_t *mutex);
 
 // condition
-PyAPI_FUNC(int) pthread_cond_init(pthread_cond_t *restrict cond,
+int pthread_cond_init(pthread_cond_t *restrict cond,
                                   const pthread_condattr_t *restrict attr);
-PyAPI_FUNC(int) pthread_cond_destroy(pthread_cond_t *cond);
-PyAPI_FUNC(int) pthread_cond_wait(pthread_cond_t *restrict cond,
+int pthread_cond_destroy(pthread_cond_t *cond);
+int pthread_cond_wait(pthread_cond_t *restrict cond,
                                   pthread_mutex_t *restrict mutex);
-PyAPI_FUNC(int) pthread_cond_timedwait(pthread_cond_t *restrict cond,
+int pthread_cond_timedwait(pthread_cond_t *restrict cond,
                                        pthread_mutex_t *restrict mutex,
                                        const struct timespec *restrict abstime);
-PyAPI_FUNC(int) pthread_cond_signal(pthread_cond_t *cond);
-PyAPI_FUNC(int) pthread_condattr_init(pthread_condattr_t *attr);
-PyAPI_FUNC(int) pthread_condattr_setclock(
+int pthread_cond_signal(pthread_cond_t *cond);
+int pthread_condattr_init(pthread_condattr_t *attr);
+int pthread_condattr_setclock(
     pthread_condattr_t *attr, clockid_t clock_id);
 
 // pthread
-PyAPI_FUNC(int) pthread_create(pthread_t *restrict thread,
+int pthread_create(pthread_t *restrict thread,
                                const pthread_attr_t *restrict attr,
                                void *(*start_routine)(void *),
                                void *restrict arg);
-PyAPI_FUNC(int) pthread_detach(pthread_t thread);
-PyAPI_FUNC(int) pthread_join(pthread_t thread, void** value_ptr);
-PyAPI_FUNC(pthread_t) pthread_self(void);
-PyAPI_FUNC(int) pthread_exit(void *retval) __attribute__ ((__noreturn__));
-PyAPI_FUNC(int) pthread_attr_init(pthread_attr_t *attr);
-PyAPI_FUNC(int) pthread_attr_setstacksize(pthread_attr_t *attr, size_t stacksize);
-PyAPI_FUNC(int) pthread_attr_destroy(pthread_attr_t *attr);
+int pthread_detach(pthread_t thread);
+int pthread_join(pthread_t thread, void** value_ptr);
+pthread_t pthread_self(void);
+int pthread_exit(void *retval) __attribute__ ((__noreturn__));
+int pthread_attr_init(pthread_attr_t *attr);
+int pthread_attr_setstacksize(pthread_attr_t *attr, size_t stacksize);
+int pthread_attr_destroy(pthread_attr_t *attr);
 
 
 // pthread_key
@@ -96,10 +96,10 @@ PyAPI_FUNC(int) pthread_attr_destroy(pthread_attr_t *attr);
 #  define PTHREAD_KEYS_MAX 128
 #endif
 
-PyAPI_FUNC(int) pthread_key_create(pthread_key_t *key,
+int pthread_key_create(pthread_key_t *key,
                                    void (*destr_function)(void *));
-PyAPI_FUNC(int) pthread_key_delete(pthread_key_t key);
-PyAPI_FUNC(void *) pthread_getspecific(pthread_key_t key);
-PyAPI_FUNC(int) pthread_setspecific(pthread_key_t key, const void *value);
+int pthread_key_delete(pthread_key_t key);
+void * pthread_getspecific(pthread_key_t key);
+int pthread_setspecific(pthread_key_t key, const void *value);
 
 #endif // Py_CPYTHON_PTRHEAD_STUBS_H
