--- a/numpy/_core/meson.build	2026-02-01 03:50:45.000000000 +0800
+++ b/numpy/_core/meson.build	2026-02-26 22:12:00.215718466 +0800
@@ -719,7 +719,7 @@
 unique_hash_cpp_args = c_args_common
 if cc.get_argument_syntax() != 'msvc'
   unique_hash_cpp_args += [
-    '-fexceptions',
+    '-fno-exceptions',
     '-fno-rtti',  # no runtime type information
   ]
 endif
--- a/numpy/_core/src/common/npy_hashtable.cpp	2026-02-01 03:50:45.000000000 +0800
+++ b/numpy/_core/src/common/npy_hashtable.cpp	2026-02-26 22:03:21.517425486 +0800
@@ -15,7 +15,6 @@
 #include "npy_hashtable.h"
 
 #include <mutex>
-#include <shared_mutex>
 
 #include "templ_common.h"
 #include <new>
@@ -112,14 +111,6 @@
         return NULL;
     }
 
-#ifdef Py_GIL_DISABLED
-    res->mutex = new(std::nothrow) std::shared_mutex();
-    if (res->mutex == nullptr) {
-        PyErr_NoMemory();
-        PyMem_Free(res);
-        return NULL;
-    }
-#endif
     return res;
 }
 
@@ -128,9 +119,6 @@
 PyArrayIdentityHash_Dealloc(PyArrayIdentityHash *tb)
 {
     PyMem_Free(tb->buckets);
-#ifdef Py_GIL_DISABLED
-    delete (std::shared_mutex *)tb->mutex;
-#endif
     PyMem_Free(tb);
 }
 
--- a/numpy/_core/src/multiarray/unique.cpp	2026-02-01 03:50:45.000000000 +0800
+++ b/numpy/_core/src/multiarray/unique.cpp	2026-02-26 22:12:23.103291300 +0800
@@ -6,9 +6,7 @@
 #include <algorithm>
 #include <complex>
 #include <cstring>
-#include <functional>
-#include <stdexcept>
-#include <unordered_map>
+#include <map>
 #include <unordered_set>
 #include <vector>
 
@@ -369,11 +367,8 @@
             int is_null = NpyString_load(alloc.allocator(), packed_string,
                                          &unpacked_strings[i]);
             if (is_null == -1) {
-                // Unexpected error. Throw a C++ exception that will be caught
-                // by the caller of unique_vstring() and converted into a Python
-                // RuntimeError.
-                throw std::runtime_error("Failed to load string from packed "
-                                         "static string.");
+                // Unexpected error.
+                std::abort();
             }
             hashset.insert(&unpacked_strings[i]);
         }
@@ -413,8 +408,8 @@
 
 
 // this map contains the functions used for each item size.
-typedef std::function<PyObject *(PyArrayObject *, npy_bool)> function_type;
-std::unordered_map<int, function_type> unique_funcs = {
+typedef PyObject *(*function_type)(PyArrayObject *, npy_bool);
+std::map<int, function_type> unique_funcs = {
     {NPY_BYTE, unique_numeric<npy_byte, hash_integer<npy_byte>, equal_integer<npy_byte>, copy_integer<npy_byte>>},
     {NPY_UBYTE, unique_numeric<npy_ubyte, hash_integer<npy_ubyte>, equal_integer<npy_ubyte>, copy_integer<npy_ubyte>>},
     {NPY_SHORT, unique_numeric<npy_short, hash_integer<npy_short>, equal_integer<npy_short>, copy_integer<npy_short>>},
@@ -504,23 +499,14 @@
     }
 
     PyObject *result = NULL;
-    try {
-        auto type = PyArray_TYPE(arr);
-        // we only support data types present in our unique_funcs map
-        if (unique_funcs.find(type) == unique_funcs.end()) {
-            result = Py_NewRef(Py_NotImplemented);
-        }
-        else {
-            result = unique_funcs[type](arr, equal_nan);
-        }
+    auto type = PyArray_TYPE(arr);
+    // we only support data types present in our unique_funcs map
+    auto it = unique_funcs.find(type);
+    if (it == unique_funcs.end()) {
+        result = Py_NewRef(Py_NotImplemented);
     }
-    catch (const std::bad_alloc &e) {
-        PyErr_NoMemory();
-        result = NULL;
-    }
-    catch (const std::exception &e) {
-        PyErr_SetString(PyExc_RuntimeError, e.what());
-        result = NULL;
+    else {
+        result = (it->second)(arr, equal_nan);
     }
     Py_DECREF(arr);
     return result;
--- a/numpy/_core/src/umath/reduction.c	2026-02-01 03:50:45.000000000 +0800
+++ b/numpy/_core/src/umath/reduction.c	2026-02-26 22:03:21.518312938 +0800
@@ -21,7 +21,7 @@
 #include "array_coercion.h"
 #include "array_method.h"
 #include "ctors.h"
-#include "refcount.h"
+#include "../multiarray/refcount.h"
 
 #include "numpy/ufuncobject.h"
 #include "lowlevel_strided_loops.h"
--- a/numpy/fft/_pocketfft_umath.cpp	2026-02-01 03:50:45.000000000 +0800
+++ b/numpy/fft/_pocketfft_umath.cpp	2026-02-26 22:03:21.518374081 +0800
@@ -35,19 +35,19 @@
                       npy_intp const *steps, void *func)
 {
     NPY_ALLOW_C_API_DEF
-    try {
+    // try {
         cpp_ufunc(args, dimensions, steps, func);
-    }
-    catch (std::bad_alloc& e) {
-        NPY_ALLOW_C_API;
-        PyErr_NoMemory();
-        NPY_DISABLE_C_API;
-    }
-    catch (const std::exception& e) {
-        NPY_ALLOW_C_API;
-        PyErr_SetString(PyExc_RuntimeError, e.what());
-        NPY_DISABLE_C_API;
-    }
+    // }
+    // catch (std::bad_alloc& e) {
+    //     NPY_ALLOW_C_API;
+    //     PyErr_NoMemory();
+    //     NPY_DISABLE_C_API;
+    // }
+    // catch (const std::exception& e) {
+    //     NPY_ALLOW_C_API;
+    //     PyErr_SetString(PyExc_RuntimeError, e.what());
+    //     NPY_DISABLE_C_API;
+    // }
 }
 
 /*
--- a/numpy/fft/pocketfft/pocketfft_hdronly.h	2024-03-22 17:53:05.000000000 +0800
+++ b/numpy/fft/pocketfft/pocketfft_hdronly.h	2026-02-26 22:03:21.518537715 +0800
@@ -161,7 +161,7 @@
   {
   // aligned_alloc() requires that the requested size is a multiple of "align"
   void *ptr = ::aligned_alloc(align,(size+align-1)&(~(align-1)));
-  if (!ptr) throw std::bad_alloc();
+  if (!ptr) std::abort();
   return ptr;
   }
 inline void aligned_dealloc(void *ptr)
@@ -171,7 +171,7 @@
   {
   align = std::max(align, alignof(max_align_t));
   void *ptr = malloc(size+align);
-  if (!ptr) throw std::bad_alloc();
+  if (!ptr) std::abort();
   void *res = reinterpret_cast<void *>
     ((reinterpret_cast<uintptr_t>(ptr) & ~(uintptr_t(align-1))) + uintptr_t(align));
   (reinterpret_cast<void**>(res))[-1] = ptr;
@@ -192,7 +192,7 @@
       {
       if (num==0) return nullptr;
       void *res = malloc(num*sizeof(T));
-      if (!res) throw std::bad_alloc();
+      if (!res) std::abort();
       return reinterpret_cast<T *>(res);
       }
     static void dealloc(T *ptr)
@@ -517,11 +517,11 @@
     const stride_t &stride_in, const stride_t &stride_out, bool inplace)
     {
     auto ndim = shape.size();
-    if (ndim<1) throw std::runtime_error("ndim must be >= 1");
+    if (ndim<1) std::abort();
     if ((stride_in.size()!=ndim) || (stride_out.size()!=ndim))
-      throw std::runtime_error("stride dimension mismatch");
+      std::abort();
     if (inplace && (stride_in!=stride_out))
-      throw std::runtime_error("stride mismatch");
+      std::abort();
     }
 
   static POCKETFFT_NOINLINE void sanity_check(const shape_t &shape,
@@ -533,8 +533,8 @@
     shape_t tmp(ndim,0);
     for (auto ax : axes)
       {
-      if (ax>=ndim) throw std::invalid_argument("bad axis number");
-      if (++tmp[ax]>1) throw std::invalid_argument("axis specified repeatedly");
+      if (ax>=ndim) std::abort();
+      if (++tmp[ax]>1) std::abort();
       }
     }
 
@@ -543,7 +543,7 @@
     size_t axis)
     {
     sanity_check(shape, stride_in, stride_out, inplace);
-    if (axis>=shape.size()) throw std::invalid_argument("bad axis number");
+    if (axis>=shape.size()) std::abort();
     }
 
 #ifdef POCKETFFT_NO_MULTITHREADING
@@ -781,7 +781,7 @@
       {
       lock_t lock(mut_);
       if (shutdown_)
-        throw std::runtime_error("Work item submitted after shutdown");
+        std::abort();
 
       ++unscheduled_tasks_;
 
@@ -1589,7 +1589,7 @@
     POCKETFFT_NOINLINE cfftp(size_t length_)
       : length(length_)
       {
-      if (length==0) throw std::runtime_error("zero-length FFT requested");
+      if (length==0) std::abort();
       if (length==1) return;
       factorize();
       mem.resize(twsize());
@@ -2398,7 +2398,7 @@
     POCKETFFT_NOINLINE rfftp(size_t length_)
       : length(length_)
       {
-      if (length==0) throw std::runtime_error("zero-length FFT requested");
+      if (length==0) std::abort();
       if (length==1) return;
       factorize();
       mem.resize(twsize());
@@ -2523,7 +2523,7 @@
     POCKETFFT_NOINLINE pocketfft_c(size_t length)
       : len(length)
       {
-      if (length==0) throw std::runtime_error("zero-length FFT requested");
+      if (length==0) std::abort();
       size_t tmp = (length<50) ? 0 : util::largest_prime_factor(length);
       if (tmp*tmp <= length)
         {
@@ -2560,7 +2560,7 @@
     POCKETFFT_NOINLINE pocketfft_r(size_t length)
       : len(length)
       {
-      if (length==0) throw std::runtime_error("zero-length FFT requested");
+      if (length==0) std::abort();
       size_t tmp = (length<50) ? 0 : util::largest_prime_factor(length);
       if (tmp*tmp <= length)
         {
@@ -2945,9 +2945,9 @@
       {
       auto nshares = threading::num_threads();
       if (nshares==1) return;
-      if (nshares==0) throw std::runtime_error("can't run with zero threads");
+      if (nshares==0) std::abort();
       auto myshare = threading::thread_id();
-      if (myshare>=nshares) throw std::runtime_error("impossible share requested");
+      if (myshare>=nshares) std::abort();
       size_t nbase = rem/nshares;
       size_t additional = rem%nshares;
       size_t lo = myshare*nbase + ((myshare<additional) ? myshare : additional);
@@ -2969,7 +2969,7 @@
       }
     void advance(size_t n)
       {
-      if (rem<n) throw std::runtime_error("underrun");
+      if (rem<n) std::abort();
       for (size_t i=0; i<n; ++i)
         {
         p_i[i] = p_ii;
@@ -3453,7 +3453,7 @@
   const stride_t &stride_in, const stride_t &stride_out, const shape_t &axes,
   int type, const T *data_in, T *data_out, T fct, bool ortho, size_t nthreads=1)
   {
-  if ((type<1) || (type>4)) throw std::invalid_argument("invalid DCT type");
+  if ((type<1) || (type>4)) std::abort();
   if (util::prod(shape)==0) return;
   util::sanity_check(shape, stride_in, stride_out, data_in==data_out, axes);
   cndarr<T> ain(data_in, shape, stride_in);
@@ -3471,7 +3471,7 @@
   const stride_t &stride_in, const stride_t &stride_out, const shape_t &axes,
   int type, const T *data_in, T *data_out, T fct, bool ortho, size_t nthreads=1)
   {
-  if ((type<1) || (type>4)) throw std::invalid_argument("invalid DST type");
+  if ((type<1) || (type>4)) std::abort();
   if (util::prod(shape)==0) return;
   util::sanity_check(shape, stride_in, stride_out, data_in==data_out, axes);
   cndarr<T> ain(data_in, shape, stride_in);
--- a/vendored-meson/meson/mesonbuild/compilers/mixins/clike.py	2026-01-01 00:03:28.000000000 +0800
+++ b/vendored-meson/meson/mesonbuild/compilers/mixins/clike.py	2026-02-26 22:03:21.518756361 +0800
@@ -84,10 +84,10 @@
                     # First occurrence of a library
                     group_start = i
             # Only add groups if there are multiple libraries.
-            if group_end > group_start >= 0:
-                # Last occurrence of a library
-                new.insert(group_end + 1, '-Wl,--end-group')
-                new.insert(group_start, '-Wl,--start-group')
+            #if group_end > group_start >= 0:
+            #    # Last occurrence of a library
+            #    new.insert(group_end + 1, '-Wl,--end-group')
+            #    new.insert(group_start, '-Wl,--start-group')
         # Remove system/default include paths added with -isystem
         default_dirs = self.compiler.get_default_include_dirs()
         if default_dirs:
--- a/vendored-meson/meson/mesonbuild/scripts/depfixer.py	2026-01-01 00:03:28.000000000 +0800
+++ b/vendored-meson/meson/mesonbuild/scripts/depfixer.py	2026-02-26 22:03:21.518831059 +0800
@@ -432,10 +432,10 @@
         if install_name_mappings:
             for old, new in install_name_mappings.items():
                 args += ['-change', old, new]
-        if args:
-            subprocess.check_call(['install_name_tool', fname] + args,
-                                  stdout=subprocess.DEVNULL,
-                                  stderr=subprocess.DEVNULL)
+        #if args:
+        #    subprocess.check_call(['install_name_tool', fname] + args,
+        #                          stdout=subprocess.DEVNULL,
+        #                          stderr=subprocess.DEVNULL)
     except Exception as err:
         raise SystemExit(err)
 
