diff --git a/Cargo.toml b/Cargo.toml
index df90031..343081c 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -49,7 +49,7 @@ hex = "0.5.0"
 
 [lib]
 name = "_pydantic_core"
-crate-type = ["cdylib", "rlib"]
+crate-type = ["cdylib"]
 
 [profile.release]
 lto = "fat"
@@ -105,3 +105,5 @@ too_many_lines = "allow"
 unnecessary_wraps = "allow"
 unused_self = "allow"
 used_underscore_binding = "allow"
+
+[workspace]
