diff --git a/Cargo.toml b/Cargo.toml
index df90031..343081c 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -49,7 +49,7 @@ hex = "0.4.3"
 
 [lib]
 name = "_pydantic_core"
-crate-type = ["cdylib", "rlib"]
+crate-type = ["cdylib"]
 
 [profile.release]
 lto = "fat"
@@ -105,3 +105,5 @@ too_many_lines = "allow"
 unnecessary_wraps = "allow"
 unused_self = "allow"
 used_underscore_binding = "allow"
+
+[workspace]
diff --git a/build.rs b/build.rs
index 41a831e..61d0dec 100644
--- a/build.rs
+++ b/build.rs
@@ -13,4 +13,5 @@ fn main() {
     }
     println!("cargo:rustc-check-cfg=cfg(specified_profile_use)");
     println!("cargo:rustc-env=PROFILE={}", std::env::var("PROFILE").unwrap());
+    println!("cargo:rustc-link-arg=-shared");
 }
